lib/filterx/filterx-object.h

```diff
@@ -173,10 +173,10 @@ filterx_object_clone(FilterXObject *self)
 }
 
 static inline gboolean
-filterx_object_map_to_json(FilterXObject *self, struct json_object **object)
+filterx_object_map_to_json(FilterXObject *self, struct json_object **object, FilterXObject **assoc_object)
 {
   if (self->type->map_to_json)
-    return self->type->map_to_json(self, object);
+    return self->type->map_to_json(self, object, assoc_object);
```

Comment:
I was wondering if we could set *assoc_object to self right here if the implementation did not do it.

It seems fragile to have to initialize that everywhere.

Reply:
I added a followup patch to do just that and with that a lot of extra *assoc_object initializations are gone.